src/EditorFeatures/Test/CodeGeneration/CodeGenerationTests.CSharp.cs

```diff
@@ -951,10 +951,46 @@ public async Task AddIndexer2()
                     parameters: Parameters(Parameter(typeof(int), "i")),
                     getStatements: "return String.Empty;",
                     isIndexer: true,
-                    options: new Dictionary<OptionKey2, object> {
-                        { CSharpCodeStyleOptions.PreferExpressionBodiedAccessors, CSharpCodeStyleOptions.NeverWithSilentEnforcement },
-                        { CSharpCodeStyleOptions.PreferExpressionBodiedIndexers, CSharpCodeStyleOptions.NeverWithSilentEnforcement },
-                    });
+                    codeGenerationOptions: new CodeGenerationOptions(
+                        options: new TestOptionSet(new Dictionary<OptionKey2, object> {
+                            { CSharpCodeStyleOptions.PreferExpressionBodiedAccessors, CSharpCodeStyleOptions.NeverWithSilentEnforcement },
+                            { CSharpCodeStyleOptions.PreferExpressionBodiedIndexers, CSharpCodeStyleOptions.NeverWithSilentEnforcement },
+                    })));
+            }
+
+            internal class TestOptionSet : OptionSet
```

Comment:
❗ Use `OptionsCollection`, and apply the options to the workspace to get an `OptionSet`.